### PR TITLE
Add JS type definitions for calculations

### PR DESCRIPTION
**Pull request by jerivas**
A continuation of #3605 and #3619

Related to 
https://github.com/sass/dart-sass/pull/1988
https://github.com/sass/sass-spec/pull/1918

### CODE REVIEW

**Review by nex3 — APPROVED** [review]
Will hold off on merging until the implementation/test PRs are also in place :sweat_smile:

File: js-api-doc/value/calculation.d.ts
Comment (jerivas) [inline js-api-doc/value/calculation.d.ts]: @nex3 looking again at this, shouldn't this be `other: Value` or even `other: any`?

File: js-api-doc/value/calculation.d.ts
Comment (jerivas) [inline js-api-doc/value/calculation.d.ts:110]: Curious as to why these are specified as getters instead of `readonly` properties?

Comment (nex3) [reply js-api-doc/value/calculation.d.ts:110]: We want to leave open the possibility for implementations to implement these using getters. Does TypeScript allow that for readonly properties? I don't actually know.

File: js-api-doc/value/calculation.d.ts
Comment (nex3) [reply js-api-doc/value/calculation.d.ts]: Yes, it probably should. It's pretty common for implementations to find small things like this that need to be updated in the proposal; feel free to lump those proposal fixes into this PR.

Comment (jerivas) [reply js-api-doc/value/calculation.d.ts]: Makes sense. This came up because the embedded host uses `other: Value`

File: js-api-doc/value/calculation.d.ts
Comment (jerivas) [reply js-api-doc/value/calculation.d.ts:110]: To test, I went with `readonly` in https://github.com/oddbird/embedded-host-node/blob/js-api-calculations/lib/src/value/calculations.ts#L113-L115 and tsc didn't complain, but I also see the value in making them getters. Will defer to you on this one

Comment (nex3) [reply js-api-doc/value/calculation.d.ts:110]: We use getters elsewhere so I think let's stick with them for consistency.

File: js-api-doc/value/calculation.d.ts
Comment (nex3) [reply js-api-doc/value/calculation.d.ts]: Actually, `other: Value` is wrong because `CalculationOperation` isn't a `Value`. I think `other: unknown` is what `Immutable.js` uses.

Comment (jerivas) [reply js-api-doc/value/calculation.d.ts]: Fixed in #3631

**Pull request merged**
